### ISSUE TITLE
Cassiopeia theme with alternative colors bug

### ISSUE THREAD

**Issue by pl71** [joomla/joomla-cms#43071]
Not sure if issue [42679](https://github.com/joomla/joomla-cms/issues/42679) is connected, but noticed
that `--link-color: #224faa;` is hardcoded in Cassiopeia. 
I've tried "Templates: Edit Style/Advanced/Colour Theme=Alternative", but links are still blue.
"\media\templates\site\cassiopeia\css\global\colors_alternative.css" has at row 4 '--link-color: #30638d;', but it stays not used IMHO.

### Steps to reproduce the issue
Have a brand new site with a Cassiopeia theme.
Switch to the "Alternative" color theme.

### Expected result
Links to use color from `colors_alternative.css`

### Actual result
Links are using color from `colors_standard.css`
![link_color](https://github.com/joomla/joomla-cms/assets/4776779/50c1b545-b672-45d0-a027-00232bc15552)

### System information (as much as possible)
J5.0.3

### Additional comments
Same behavior with the child template.
EDIT: Tried also [totally different colors](https://github.com/joomla/joomla-cms/pull/35917), but `link-color` has to be defined in user.css to work.

**Comment by ReLater** [comment]
`colors_alternative.css` defines the link color as blue. Other words: #30638d is a kind of blue.

![grafik](https://github.com/joomla/joomla-cms/assets/20780646/7bb565a1-11ba-4df0-be39-19a002bc1382)

![grafik](https://github.com/joomla/joomla-cms/assets/20780646/6caa41a6-71e4-4146-ad63-d2d5e7f483dc)

**Comment by ReLater** [comment]
AH, sorry! Yes you're right. Wrong color. I think colors_alternative.css should be loaded AFTER template.css.

![grafik](https://github.com/joomla/joomla-cms/assets/20780646/241a1b57-ef65-44eb-9a2c-d50814405120)

Looks like it's the same in J!5.1 (but I'm tired ;-) )

**Comment by pl71** [comment]
colors_alternative should modify template.css in order to reduce css files on demand.

**Comment by ReLater** [comment]
Off Topic. Open a [discussion](https://github.com/joomla/joomla-cms/discussions)?

> colors_alternative should modify template.css in order to reduce css files on demand.

I compile my CSS files from SASS/Scss. Also in Cassiopeia child templatess. Then things like that (removing redundant CSS code and combining) are "easily" possible.

But I think your expectation is too much that whenever changing a colors scheme template.css, template.min.css, template.min.css.gz are newly generated. From my point of view it would also obfuscate how colors_xyz.css work. It would become obscure science in this core template that should be as minimalistic as possible. From my point of view ;-)

**Comment by richard67** [comment]
Closing as having a pull request. Please test #43084 . Thanks in advance.

**Comment by ReLater** [comment]
I have closed my pr because:

> Closing as this has already been fixed in https://github.com/joomla/joomla-cms/pull/42794/files in another way **but not ported to Joomla 5 yet**.

I don't know how to port that.

**Comment by pl71** [comment]
@ReLater Not quite sure, but these variables should go **before** `template.css`
`template.css` is using them, at least `--link-color`:
3333: 
```
.btn-link {
    --btn-font-weight: 400;
    --btn-color: var(--link-color);
    ....
```

**Comment by richard67** [comment]
> I have closed my pr because:
> 
> > Closing as this has already been fixed in https://github.com/joomla/joomla-cms/pull/42794/files in another way **but not ported to Joomla 5 yet**.
> 
> I don't know how to port that code.

@ReLater See my comment in your closed PR. The fix should have been ported to the 5.1-dev branch.

**Comment by ReLater** [comment]
> Not quite sure, but these variables should go before template.css
template.css is using them, at least --link-color:

Nope. Because it's a :root{} rule. See image:

![grafik](https://github.com/joomla/joomla-cms/assets/20780646/498c1a95-fd6f-43e7-a790-041bfa276997)
